streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableFilter.java

```diff
@@ -28,7 +28,7 @@
     private final Predicate<K, V> predicate;
     private final boolean filterNot;
 
-    private boolean sendOldValues = false;
+    private boolean sendOldValues = true;
```

Comment:
My concern is that, the overhead of requesting the source KTable to be materialized (i.e. creating a state store, and sending the {old -> new} pair instead of the new value only) may be over-whelming compared with its potential benefits of reducing the downstream traffic.